### PR TITLE
fixed issue with Kooapp for false positive #1493

### PR DESCRIPTION
**Pull request by keder-code-hash**
- Fixed The issue with False positive for KOOAPP
- Updated URL for searching

### CODE REVIEW

**Comment by sdushantha** [comment]
This PR does not solve the false positive. You are using the the search query endpoint and that endpoint does not return an `errorMsg` of "Invalid Profile"

**Comment by keder-code-hash** [comment]
> This PR does not solve the false positive. You are using the the search query endpoint and that endpoint does not return an `errorMsg` of "Invalid Profile"

Hey, Siddharth I observed the Kooapp API call. The/profile/{} won't give us the actual profile and There is no way of checking a particular API call for a particular platform in sherlock. I mean for kooapp,  I can make an API call to check the existence of any user and then we can return his profile details. But I think this type of facility is not available for this project. Can we add that by changing the design pattern to adapter one?
Or else in the data.json, we can associate a function for checking too for a particular platform.

**Comment by keder-code-hash** [comment]
Please let me know If any other way I can do it. Though for many other profiles we have to make a call for checking the existence of a user. In sherlock.py it is being checked whether the response is returning the correct status or not. Though many platforms do not support the status code properly such as Kooapps. The just do it their own way.

**Comment by keder-code-hash** [comment]
If we can update the pattern a little bit it would be more feasible I think. What do u say?

**Comment by sdushantha** [comment]
I did some digging and found an API endpoint: `https://www.kooapp.com/apiV1/users/handle/USERNAME_HERE/valid`

```console
$ curl https://www.kooapp.com/apiV1/users/handle/b23432ob/valid
{"status":"SUCCESS","message":"handle b23432ob is available"}

$ curl https://www.kooapp.com/apiV1/users/handle/bob/valid
{"status":"ERROR","message":"handle bob already taken"}
```

Maybe you could use this

**Comment by keder-code-hash** [comment]
> I did some digging and found an API endpoint: `https://www.kooapp.com/apiV1/users/handle/USERNAME_HERE/valid`
> 
> ```
> $ curl https://www.kooapp.com/apiV1/users/handle/b23432ob/valid
> {"status":"SUCCESS","message":"handle b23432ob is available"}
> 
> $ curl https://www.kooapp.com/apiV1/users/handle/bob/valid
> {"status":"ERROR","message":"handle bob already taken"}
> ```
> 
> Maybe you could use this

Yeah. Thanks for your reply.
I will change it.

**Comment by keder-code-hash** [comment]
I have added probeUrl as the checking one.
pl check it once. @sdushantha

File: sherlock/resources/data.json [hunk 1]
```diff
@@ -2447,12 +2447,13 @@
     "username_unclaimed": "noonewouldeverusethis77777"
   },
   "koo": {
-    "errorMsg": "Invalid Profile",
+    "errorMsg": "handle {} is available",
```
Comment (sdushantha) [inline sherlock/resources/data.json:2450]: Two minor issues here:
1. "is available" is not an `errorMsg`. "already taken" would be an error message as that is the message for when a username is not found
2. I see that you have used `{}` in the `errorMsg`. Sherlock won't format the string with the username. So it is better to just use `already taken` as the `errorMsg`.

In other words, replace the value for `errorMsg` with `already taken`

**Review by sdushantha — CHANGES_REQUESTED** [review]

File: sherlock/resources/data.json (hunk 1 above)
Comment (sdushantha) [reply sherlock/resources/data.json:2450]: Sorry my bad. It should just be `is available`

**Review by sdushantha — APPROVED** [review]

**Pull request merged**